### PR TITLE
Integration tests tracing

### LINKED ISSUE #977: Integration tests for tracing
It would be good to add integration tests to ensure that tracing works properly. From a high-level point of view this means that:

- Traces get pushed to Jaeger both from the engine and from any other component in the inference graph.
- Traces get pushed in the right format, with the engine wrapping all inner requests inside the inference graph. 
- Traces work both with REST and gRPC.

An example of what can be considered a properly working trace on Jaeger's UI can be seen below:

Working Trace

### PR DESCRIPTION
**Pull request by adriangonz**
Fixes #977 

## Changelog
- Update `retrying` to `tenacity` (`retrying` is [now unmantained](https://julien.danjou.info/python-tenacity/)).
- Add K8s configs to spin up Jaeger in integration tests.
- Add helper methods to fetch traces from [Jaeger's internal REST API](https://www.jaegertracing.io/docs/1.13/apis/#http-json-internal).
- Add simple test to assert traces get reported correctly when using the REST API.

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **adriangonz**

If they are not already assigned, you can assign the PR to them by writing `/assign @adriangonz` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Wed Feb 19 17:01:19 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=2

**Comment by seldondev** [comment]
Wed Feb 19 17:01:27 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=1

**Comment by seldondev** [comment]
Wed Feb 19 17:01:33 UTC 2020
The logs for [integration] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=3

**Comment by seldondev** [comment]
@adriangonz: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 2429329821b5183a8945bf555c0b7464f3eaf08a | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Thu Feb 20 14:41:34 UTC 2020
The logs for [integration] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=6

**Comment by seldondev** [comment]
Thu Feb 20 14:42:50 UTC 2020
The logs for [pr-build] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=4

**Comment by seldondev** [comment]
Thu Feb 20 14:42:58 UTC 2020
The logs for [lint] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=5

**Comment by seldondev** [comment]
@adriangonz: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 0b6eecefe9f3645cb6f27ad5e3a6d122e59fcf8a | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Thu Feb 20 15:49:18 UTC 2020
The logs for [integration] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=7

**Comment by seldondev** [comment]
@adriangonz: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 0b6eecefe9f3645cb6f27ad5e3a6d122e59fcf8a | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Mon Feb 24 13:21:18 UTC 2020
The logs for [pr-build] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=8

**Comment by seldondev** [comment]
Mon Feb 24 13:21:35 UTC 2020
The logs for [lint] [9] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/9.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=9

**Comment by seldondev** [comment]
Mon Feb 24 13:22:09 UTC 2020
The logs for [integration] [10] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/10.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=10

**Comment by seldondev** [comment]
Mon Feb 24 14:56:47 UTC 2020
The logs for [integration] [11] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/11.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=11

**Comment by seldondev** [comment]
@adriangonz: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 9df5b49d6cbee78da315c36e6f49b6135fab71b5 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Tue Feb 25 11:59:22 UTC 2020
The logs for [lint] [13] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/13.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=13

**Comment by seldondev** [comment]
Tue Feb 25 11:59:25 UTC 2020
The logs for [pr-build] [12] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/12.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=12

**Comment by seldondev** [comment]
Tue Feb 25 11:59:44 UTC 2020
The logs for [integration] [14] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/14.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=14

**Comment by seldondev** [comment]
Tue Feb 25 13:27:32 UTC 2020
The logs for [integration] [15] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1464/15.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1464 --build=15

File: testing/resources/kind_config.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  - role: control-plane
+  - role: worker
+    extraPortMappings:
+      - containerPort: 30080
+        hostPort: 8003
+      - containerPort: 31280
+        hostPort: 8004
+kubeadmConfigPatches:
+  - |
+    apiVersion: kubelet.config.k8s.io/v1beta1
```
Comment (axsaucedo) [inline testing/resources/kind_config.yaml]: Not relevant to this PR specifically but just realised we should try to move away from v1beta1 as it was dropped from v1.16 onwards

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/SeldonIO/seldon-core/pull/1464#issuecomment-591914315" title="LGTM">axsaucedo</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**